### PR TITLE
Update QCut

### PR DESCRIPTION
**Pull request by JMuff22**
Hi @JooNiv , the project is great. Here are some changes to the project structure and documentation to clean it up a little.

## What's changed

- Added github actions workflows for building the documentation, running tests and testing that the documentation can be built
  - This means that the docs can be built and hosted on github pages for example
  - This also reduces the overall size of the code base (makes cloning quicker) as the built documentation no longer lives inside the repo. 
  - The tests will currently fail. Could you help me in fixing them in this PR?
- Fixed all warnings from sphinx when building the documentation
  - This was mostly formatting and underline issues
 - Fixed spelling mistakes where I saw them
 - Documentation now uses Sphinx Book theme instead of read the docs theme 
   - This enables jupyter notebooks to be rendered in the docs
   - Added a notebooks folder and RST file for this
  -  Added `pyproject.toml`and deprecated `setup.py`
    - This allows `ruff.toml`and `setup.py`to be integrated into 1 file. Also optional dependencies have been added. 
  - Relative imports are now absolute imports 
    - Which reduces the potential for circular import errors 
  - Images are now located under the `_static`folder. No need to have to folders for static content
    - Added your QCut logo and favicon 
  

The README now contains the information to build the docs. 

## To-do

- workflow to publish to PyPI
- fix tests
- Add linting to CI pipeline
- Fix versioning and add a changelog

### CODE REVIEW

**Comment by JooNiv** [comment]
Ignore the failed docs build on commit 463b01d. It only failed because I pushed a new commit before the build finished causing it to fail

**Comment by JMuff22** [comment]
@JooNiv can you check that github actions can work in forks? Then we can test the workflows in this PR.

File: .github/workflows/tag_and_release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  check_version:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check version in changelog
+        run: bash ./tag-from-pipeline.sh verify_changelog_version
```
Comment (JMuff22) [inline .github/workflows/tag_and_release.yml:22]: I believe this needs access to the github token? e.g 

```
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Publish
```
Comment (JMuff22) [inline .github/workflows/publish.yml:1]: This workflow can either be renamed to "Publish docs" or be combined with the `publish_to_pypi.yml` file

File: .github/workflows/tag_and_release.yml (hunk 1 above)
Comment (JMuff22) [reply .github/workflows/tag_and_release.yml:22]: Also, there is no method of creating the tag.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (JooNiv) [reply .github/workflows/publish.yml]: Renamed to Publish docs

File: .github/workflows/tag_and_release.yml (hunk 1 above)
Comment (JooNiv) [reply .github/workflows/tag_and_release.yml:22]: Added github token to the enviroment and method for creating a tag and a release

**Pull request merged**